staging/src/k8s.io/client-go/transport/dial.go

```diff
@@ -0,0 +1,36 @@
+package transport
+
+import (
+	"net"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/obeattie/tcp-failfast"
+)
+
+// failFastDial wraps a dial function and sets up a TCP "user timeout" on the
+// connection. This is useful because we want to detect dead connections to
+// the apiserver quickly, and default kernel parameters mean this usually takes
+// 15+ minutes.
+func failFastDial(d func(network, addr string) (net.Conn, error)) func(network, addr string) (net.Conn, error) {
+	const timeout = 7 * time.Second
```

Comment:
We have `transport. Config`. Isn't that a good place for that? Or is there any need to change that per dial attempt?